File: backend/src/test/java/org/luxons/sevenwonders/game/data/GameDefinitionTest.java
package org.luxons.sevenwonders.game.data;

import org.junit.Test;
import org.luxons.sevenwonders.game.Game;
import org.luxons.sevenwonders.game.api.CustomizableSettings;

import static org.junit.Assert.assertNotNull;

public class GameDefinitionTest {

    @Test
    public void successfulGameInit() throws Exception {
        GameDefinition gameDefinition = new GameDefinitionLoader().getGameDefinition();
        assertNotNull(gameDefinition);

        Game game = gameDefinition.initGame(0, new CustomizableSettings(), 7);
        assertNotNull(game);
    }
}

File: sw-engine/src/test/kotlin/org/luxons/sevenwonders/engine/data/GameDefinitionTest.kt
package org.luxons.sevenwonders.engine.data

import org.junit.Test
import org.luxons.sevenwonders.model.Settings
import org.luxons.sevenwonders.model.wonders.deal
import kotlin.test.assertEquals
import kotlin.test.assertNotNull

class GameDefinitionTest {

    @Test
    fun successfulGameInit() {
        val gameDefinition = GameDefinition.load()
        assertNotNull(gameDefinition)
        assertEquals(3, gameDefinition.minPlayers)
        assertEquals(7, gameDefinition.maxPlayers)

        val wonders = gameDefinition.allWonders.deal(7)
        val game = gameDefinition.createGame(0, wonders, Settings())
        assertNotNull(game)
    }
}
